### ISSUE TITLE
Exception: Request is not authentic.

### ISSUE THREAD

**Issue by AlbertCalvet** [ishahrier/ExicoShopifyFramework#15]
Hi!

I'm just testing the solution to see if I can make it the starter point for a Shopify integration.

I'd swear I followed all steps from the **quick introduction** wiki (including OFC adding my own partner developer API Key and secret in the appsettings.Development.json file) but I still get this exception while trying to install the app from my developer shop.

Here's the stack trace:

`System.Exception: Request is not authentic.
   at Exico.Shopify.Web.Core.Controllers.BaseControllers.ABaseShopifyController.Handshake(String shop)
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)`

Any idea why? Does this still work in 2020?
Thanks!!

**Comment by ishahrier** [comment]
I think updating the shopifysharp package would eliminate this error. I do not have time right now..may be on the weekend i will give it a go. i am now working on updating  the framework to aspnet core 3.0 and adding angular to it.

**Comment by david-mclean** [comment]
@AlbertCalvet  We ran into this issue a couple time in the past. I believe the API keys in the database were not correct. 

When you first run the application it will put the keys from the config into the database, If you change the keys in your config I dont believe the DB gets reliably updated to the same.

**Comment by AlbertCalvet** [comment]
@david-mclean You are right! So I had a little bit of success by directly opening the github project and setting the eXico.Shopify.Web.Core project as the default starting project and running that one. Now that you mention it, maybe in this case I hit the play button AFTER changing the json file, while on my own project from the template I did it before so that's why I got the default settings. Seems a little bit confusing but now I'm up and running to start playing with it. Thanks a lot!

@ishahrier thanks also for the fast reply, seems like this is solved now.
